### PR TITLE
Portals access permission for automation-dashboard

### PR DESCRIPTION
**Pull request by aminelouati**
In this PR, 
- we implement the control of portals access permission for automation-dashboard
- adapt portal links on the left side navbar

### CODE REVIEW

File: app/scripts/main.js [hunk 1]
```diff
@@ -142,14 +177,23 @@ mainModule.config(function ($translateProvider, $translatePartialLoaderProvider)
 
 // --------------- Controllers -----------------
 
-mainModule.controller('mainController', function ($http, $scope, $rootScope, $state, $location, $interval, $translate) {
+mainModule.controller('mainController', function ($http, $scope, $rootScope, $state, $location, $interval, $translate, permissionService, $q) {
```
Comment (medou-boushab) [inline app/scripts/main.js]: change the order permissionService, $q => $q, permissionService

File: Gruntfile.js [hunk 2]
```diff
@@ -304,42 +303,42 @@ module.exports = function (grunt) {
                                 for (var key in subviewsDefinition) {
                                     if (subviewsDefinition[key].isSubMenuTitle) {
                                         if(subviewsDefinition[key].name == 'Analytics'){
-                                            result += '\n<li ui-sref-active="active">'
+                                            result += '\n<li id="'+subviewsDefinition[key].name+'" ui-sref-active="active" ng-show="showParentPortal(\''+subviewsDefinition[key].name+'\')">'
```
[the remark this answers is not in the document]
Comment (amouhoub) [reply Gruntfile.js]: @medou-boushab In this particular case, we need ng-show to make some jquery selectors to show/hide submenus : See the function `showParentPortal( )`

File: app/scripts/main.js [hunk 3]
```diff
@@ -142,14 +177,23 @@ mainModule.config(function ($translateProvider, $translatePartialLoaderProvider)
...
         $scope.contextPosition = '';
+        $scope.firstAccessiblePortal = '';
+        $scope.portalsAccessPermission = {};
+        if(getSessionId()){
+            $scope.determineFirstAuthorizedPortalAndAllPortalsAccessPermission();
+        }
+    };
+
+    $scope.showParentPortal = function(id){
+        return angular.element(document.getElementById(id)).next().hasClass('childPortal');
```
Comment (amouhoub) [inline app/scripts/main.js]: @medou-boushab it's this function that requires ng-show instead of ng-if

File: app/scripts/main.js [hunk 4]
```diff
@@ -142,14 +177,23 @@ mainModule.config(function ($translateProvider, $translatePartialLoaderProvider)
...
 
     this.$onInit = function () {
         $scope.main.userName = localStorage['pa.login'];
         $scope.startRegularCheckSession();
         $scope.contextDisplay = false;
         // contextPosition enables directive to specify where the context menu was opened
         $scope.contextPosition = '';
+        $scope.firstAccessiblePortal = '';
+        $scope.portalsAccessPermission = {};
+        if(getSessionId()){
```
Comment (amouhoub) [inline app/scripts/main.js]: Add extra space after `if`

File: app/scripts/main.js [hunk 5]
```diff
@@ -191,6 +235,62 @@ mainModule.controller('mainController', function ($http, $scope, $rootScope, $st
         }
     }
 
+    $scope.checkPortalAccessPermission = function(url){
+        var portal = url.substring(url.lastIndexOf("/") + 1);
+        permissionService.getPortalAccessPermission(portal)
+            .success(function (response) {
```
Comment (amouhoub) [inline app/scripts/main.js]: 1) .success and .error are deprecated. --> Use then, catch
2) it's better to handle errors inside services and only make the logic-in-case-of-success-request in the controller. This is how it's recommended to do so:

In the service do :
```
return $http.get(.....).then(function(response){
//... here do some parsing if required or other stuff then do 
return response.data // or whatever variable for the response
})
.catch(function(response){
// console.error("whatever", response)
});
```
and inside your function in the controller just do:
```
permissionService.getPortalAccessPermission(porta).then(function response){
// do you logic with response (not response.data as the service already returns response.data
});
```

File: app/scripts/main.js [hunk 6]
```diff
@@ -191,6 +235,62 @@ mainModule.controller('mainController', function ($http, $scope, $rootScope, $st
...
+    $scope.determineFirstAuthorizedPortalAndAllPortalsAccessPermission = function() {
+        var automationDashboardPortals = {};
+        $state.get().forEach(function (item) {
+            if(item.name && item.name !== 'login' && item.name !== 'portal'){
+                automationDashboardPortals[item.url.substring(1)] = item.name;
+                $scope.portalsAccessPermission[item.url.substring(1)] = false;
+            }
+        });
+        var portals = Object.keys(automationDashboardPortals);
+        permissionService.getPortalsAccessPermission(portals)
```
Comment (amouhoub) [inline app/scripts/main.js]: Same comment as above

**Pull request merged**